src/Magnum/Trade/Test/MeshDataTest.cpp

```diff
@@ -2500,6 +2522,90 @@ void MeshDataTest::objectIdsIntoArrayInvalidSize() {
+    auto weightsView = Containers::arrayCast<T>(vertexData);
+    weightsView[0] = T::pad(Math::Vector4<typename T::Type>{Math::pack<typename T::Type>(1.0f), 0, Math::pack<typename T::Type>(1.0f), Math::pack<typename T::Type>(0.8)});
+    weightsView[1] = T::pad(Math::Vector4<typename T::Type>{0, Math::pack<typename T::Type>(1.0f), 0, Math::pack<typename T::Type>(0.4f)});
+
+    MeshData data{MeshPrimitive::Points, std::move(vertexData), {MeshAttributeData{MeshAttribute::Weights,
+        /* Assuming the normalized enum is always after the non-normalized */
+        VertexFormat(UnsignedInt(Implementation::vertexFormatFor<T>()) + 1),
+        weightsView}}};
+    CORRADE_COMPARE_AS(data.weightsAsArray(), Containers::arrayView<Vector4>({
+        Vector4::pad(Math::Vector<T::Size, Float>::pad(Vector4{1.0f, 0.0f, 1.0f, 0.8})),
+        Vector4::pad(Math::Vector<T::Size, Float>::pad(Vector4{0.0f, 1.0f, 0.0f, 0.4f}))
+    }), TestSuite::Compare::Container);
+}
+
+void MeshDataTest::weightsIntoArrayInvalidSize() {
+    #ifdef CORRADE_NO_ASSERT
+    CORRADE_SKIP("CORRADE_NO_ASSERT defined, can't test assertions");
+    #endif
+
+    Containers::Array<char> vertexData{3*sizeof(Vector4)};
+    MeshData data{MeshPrimitive::Points, std::move(vertexData), {MeshAttributeData{MeshAttribute::Weights, Containers::arrayCast<Vector4>(vertexData)}}};
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    Vector4 destination[2];
+    data.weightsInto(destination);
+    CORRADE_COMPARE(out.str(),
+        "Trade::MeshData::weightsInto(): expected a view with 3 elements but got 2\n");
+}
+
+template<class T> void MeshDataTest::jointIdsAsArray() {
+    setTestCaseTemplateName(NameTraits<T>::name());
+    typedef typename T::Type U;
+
+    Containers::Array<char> vertexData{3*sizeof(T)};
+    auto joinIdsView = Containers::arrayCast<T>(vertexData);
+    joinIdsView[0] = {U(0), U(1), U(2), U(3)};
+    joinIdsView[1] = {U(4), U(5), U(6), U(7)};
+    joinIdsView[2] = {U(8), U(9), U(10), U(11)};
+
+    MeshData data{MeshPrimitive::Points, std::move(vertexData), {MeshAttributeData{MeshAttribute::JointIds, joinIdsView}}};
+    CORRADE_COMPARE_AS(data.jointIdsAsArray(), Containers::arrayView<Vector4ui>({
+        {0, 1, 2, 3}, {4, 5, 6, 7}, {8, 9, 10, 11},
+    }), TestSuite::Compare::Container);
+}
+
+void MeshDataTest::jointIdsIntoArrayInvalidSize() {
+    #ifdef CORRADE_NO_ASSERT
+    CORRADE_SKIP("CORRADE_NO_ASSERT defined, can't test assertions");
+    #endif
+
+    Containers::Array<char> vertexData{3*sizeof(Vector4ui)};
+    MeshData data{MeshPrimitive::Points, std::move(vertexData), {MeshAttributeData{MeshAttribute::JointIds, Containers::arrayCast<Vector4ui>(vertexData)}}};
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    Vector4ui destination[2];
+    data.jointIdsInto(destination);
+    CORRADE_COMPARE(out.str(),
+        "Trade::MeshData::jointIdsInto(): expected a view with 3 elements but got 2\n");
+}
+
```

Comment:
The `attributeNotFound()` and `implementationSpecificVertexFormatWrongAccess()` tests need to be extended to check asserts in the new accessors.